File: Herencia/ejercicio 2/Python/Electrodomestico.py
class Electrodomestico:
    def __init__(self, precioBase, color, consumoEnergico, peso):
        self.precioBase = precioBase
        self.color = color
        self.consumoEnergico = consumoEnergico.upper()
        self.peso = peso
    def precioFinal(self):
        precio = self.precioBase
        consumoPrecio = {"A": 100, "B": 80, "C": 60, "D": 50, "E": 30, "F": 10}
        precio += consumoPrecio.get(self.consumoEnergico, 0)
        if self.peso <= 19:
            precio += 10
        elif self.peso <= 49:
            precio += 50
        elif self.peso <= 79:
            precio += 80
        else:
            precio += 100
        return precio

File: Herencia/ejercicio 2/Java/Electrodomestico.java
class Electrodomestico {
    public double precioBase;
    public String color;
    public char consumoEnergetico;
    public double peso;
    public Electrodomestico(double precioBase, String color, char consumoEnergetico, double peso) {
        this.precioBase = precioBase;
        this.color = color;
        this.consumoEnergetico = Character.toUpperCase(consumoEnergetico);
        this.peso = peso;
    }
    public double precioFinal() {
        double precio = precioBase;
        switch (consumoEnergetico) {
            case 'A': precio += 100; break;
            case 'B': precio += 80; break;
            case 'C': precio += 60; break;
            case 'D': precio += 50; break;
            case 'E': precio += 30; break;
            case 'F': precio += 10; break;
        }
        if (peso <= 19) precio += 10;
        else if (peso <= 49) precio += 50;
        else if (peso <= 79) precio += 80;
        else precio += 100;

        return precio;
    }
}